uaclient/contract.py

```diff
@@ -288,5 +288,10 @@ def request_updated_contract(
 def get_available_resources(cfg) -> "List[Dict]":
     """Query available resources from the contrct server for this machine."""
     client = UAContractClient(cfg)
-    resources = client.request_resources()
+    try:
+        resources = client.request_resources()
+    except util.UrlError as exc:
```

Comment:
Should we do this error handling further up the stack, so that _any_ `util.UrlError`s that would show a traceback end up reporting `status.MESSAGE_CONNECTIVITY_ERROR` to users?

Reply:
@OddBloke @powersj:   I *think* we take url error handling up a level. Though I'm not certain what messaging we *should* be doing then in the case of ua enable|disable|refresh which already could have static failure messages.   Should we supplement those messages by first emitting the connectivity error message followed by the current enable failure message?    
Examples below:

failure to attach due to connectivity:
 
Here are some examples that I think we probably need  to cleanup

- ua enable <someinapplicableservice> with network connectivity 

```
# ua enable fips
One moment, checking your subscription first
This subscription is not entitled to FIPS.
See: sudo ua status or https://ubuntu.com/advantage
Failed to connect to authentication server
Check your Internet connection and try again

# ua enable livepatch
One moment, checking your subscription first
Cannot install Livepatch on a container
Failed to connect to authentication server
Check your Internet connection and try again

```

- ua refresh on network connectivity
```
# ua refresh
Unable to refresh your subscription
```

Reply:
I don't think we need to decide that for this change to land; regardless of what those commands' behaviour is, I still think we want a top-level handler for `util.UrlError` for the cases that we aren't currently handling.

Reply:
Agreed. I've moved this error handling up into uaclient/cli and added unit test coverage for this failure path